File: PythonEx1/Ex1.py
class Carro:
    def __init__(self, marca, modelo, ano):
        self.marca = marca
        self.modelo = modelo
        self.ano = ano

    def exibir_detalhes(self):
        print(f"Marca: {self.marca}, Modelo: {self.modelo}, Ano: {self.ano}")

# Instanciando três objetos da classe Carro
carro1 = Carro("Toyota", "Corolla", 2020)
carro2 = Carro("Ford", "Mustang", 2021)
carro3 = Carro("Chevrolet", "Camaro", 2022)

# Exibindo os detalhes de cada carro
carro1.exibir_detalhes()
carro2.exibir_detalhes()
carro3.exibir_detalhes()

File: Ex1.java
package aa;

class Carro {
    private String marca;
    private String modelo;
    private int ano;
    private int velocidade;

    public Carro(String marca, String modelo, int ano) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.velocidade = 0; 
    }

    public void acelerar(int incremento) {
        if (incremento > 0) {
            velocidade += incremento;
            System.out.println("Acelerando: " + incremento + " km/h");
        } else {
            System.out.println("Incremento deve ser positivo.");
        }
    }

    public void frear(int decremento) {
        if (decremento > 0) {
            velocidade -= decremento;
            if (velocidade < 0) {
                velocidade = 0;
            }
            System.out.println("Freando: " + decremento + " km/h");
        } else {
            System.out.println("Decremento deve ser positivo.");
        }
    }

    public void exibirVelocidade() {
        System.out.println("Velocidade atual: " + velocidade + " km/h");
    }

    public void exibirDetalhes() {
        System.out.println("Marca: " + marca + ", Modelo: " + modelo + ", Ano: " + ano);
    }

    public static void main(String[] args) {
        Carro carro1 = new Carro("Toyota", "Corolla", 2020);
        carro1.exibirDetalhes();
        carro1.acelerar(50);
        carro1.exibirVelocidade();
        carro1.frear(20);
        carro1.exibirVelocidade();
        carro1.frear(40); 
        carro1.exibirVelocidade();
    }
}
